src/schedule/zephyr_dp_schedule_application.c

```diff
@@ -19,6 +20,174 @@
 
 #include "zephyr_dp_schedule.h"
 
+LOG_MODULE_DECLARE(dp_schedule, CONFIG_SOF_LOG_LEVEL);
+extern struct tr_ctx dp_tr;
+
+/* Synchronization semaphore for the scheduler thread to wait for DP startup */
+#define DP_SYNC_INIT(i, _)	Z_SEM_INITIALIZER(dp_sync[i], 0, 1)
+#define DP_SYNC_INIT_LIST	LISTIFY(CONFIG_CORE_COUNT, DP_SYNC_INIT, (,))
+static STRUCT_SECTION_ITERABLE_ARRAY(k_sem, dp_sync, CONFIG_CORE_COUNT) = { DP_SYNC_INIT_LIST };
+
+/* TODO: make this a shared kernel->module buffer for IPC parameters */
+static uint8_t ipc_buf[4096];
+
+struct ipc4_flat {
+	unsigned int cmd;
+	int ret;
+	union {
```

Comment:
btw, is this to align with ipc objects ?

Reply:
@lgirdwood no, this is just to have storage for any possible IPC type